app/controllers/chargeback_controller.rb

```diff
@@ -338,6 +389,56 @@ def cb_assign_field_changed
     end
   end
 
+  # Add a new tier at the end
+  def cb_tier_add
+    @edit = session[:edit]
+    i = params[:id]
+    ii = i.to_i
+    @sb[:num_tiers][ii] =
+        @edit[:new][:details][ii][:chargeback_tiers].to_a.length if @edit[:new][:details][ii][:chargeback_tiers]
+    @sb[:num_tiers][ii] = 1 unless @sb[:num_tiers][ii] || @sb[:num_tiers][ii] == 0
+    @sb[:num_tiers][ii] += 1
+    pos = @sb[:num_tiers][ii] - 2
+    n = pos + 1
+    @edit[:new][:tiers][ii][n] = {}
+    @edit[:new][:tiers][ii][n][:start] = @edit[:new][:tiers][ii][pos][:end]
+    @edit[:new][:tiers][ii][n][:end] = Float::INFINITY
+    @edit[:new][:tiers][ii][n][:fix_rate] = 0.0
+    @edit[:new][:tiers][ii][n][:var_rate] = 0.0
+    render :update do |page|
+      page.replace("rate_detail_row_#{i}_0", :partial => "tier_first_row") # Update the first row to change the colspan
+      page.insert_html(:after, "rate_detail_row_#{i}_#{pos}", :partial => "tier_row") # Insert the new tier after the last one
+      page << javascript_for_miq_button_visibility(true)
+    end
+  end
+
+  # Remove the selected tier
+  def cb_tier_remove
+    @edit = session[:edit]
+    id = params[:id]
+    i = id.split("-")[0]
+    j = id.split("-")[1]
```

Comment:
Ok, thanks for the tip, i and j are the detail and tier indexes to locate which tier we want to remove. I'll change the names and add some comments.